backend/onyx/document_index/vespa/indexing_utils.py

```diff
@@ -248,3 +252,84 @@ def clean_chunk_id_copy(
         }
     )
     return clean_chunk
+
+
+def _does_doc_exist_in_vespa(
+    doc_id: str,
+    index_name: str,
+    http_client: httpx.Client,
+) -> bool:
+    """
+    Checks whether there's a chunk/doc matching doc_id in Vespa using YQL.
+    """
+    encoded_doc_id = urllib.parse.quote(doc_id)
+
+    # Construct the URL with YQL query
+    url = (
+        f"{SEARCH_ENDPOINT}"
+        f'?yql=select+*+from+sources+{index_name}+where+document_id+contains+"{encoded_doc_id}"'
+        "&hits=0"
+    )
+
+    logger.debug(f"Checking existence for doc_id={doc_id} with URL={url}")
+    resp = http_client.get(url)
+
+    if resp.status_code == 200:
```

Comment:
The flow here is probably easier to follow if we treat 200 as the happy path and the other paths as exceptions/failures.

Reply:
This was definitely the intention- is there a way we can make it clearer that 200 is the happy path in the existing code?

Reply:
Could we try:

if 404 ... return
if not 200 ... return

# 200 - happy path

Reply:
+1 to this pattern!